### PR TITLE
AG-292: Improve table sort, display, and export

### PR DESCRIPTION
**Pull request by JessterB**
* Fix similar genes table's druggability column sorts
* Export CSVs with column display names rather than raw field names
* Similar genes table's Nominated Target column no longer displays mixed types
* Prune and rename GeneInfo display fields
* Set consistent default display values for missing table data
* Fix display issue with missing alias[] in overview

### CODE REVIEW

File: src/app/genes/gene-overview/gene-overview.component.ts [hunk 1]
```diff
@@ -466,7 +466,7 @@ export class GeneOverviewComponent implements OnInit, OnDestroy, AfterContentChe
     }
 
     getAlias(): string {
-        if (this.geneInfo.alias.length > 0) {
+        if (this.geneInfo.alias && this.geneInfo.alias.length > 0) {
```
Comment (JessterB) [inline src/app/genes/gene-overview/gene-overview.component.ts:469]: This is a tiny bug fix that didn't deserve it's own PR, it's entirely unrelated to the rest of these changes...

File: src/app/genes/gene-similar/gene-similar.component.html [hunk 2]
```diff
@@ -75,9 +75,9 @@ <h1>Similar Genes</h1>
         <ng-template pTemplate="body" let-rowData let-columns="columns">
             <tr [pSelectableRow]="rowData">
                 <td *ngFor="let col of columns">
-                    {{colFinalValue(rowData, col.field, col.subfield)}}
+                    {{colFinalValue(rowData, col.field)}}
```
Comment (JessterB) [inline src/app/genes/gene-similar/gene-similar.component.html:78]: I changed genes-similar to be more like genes-list since the latter was in better shape; it might be easier to look at genes-list first.

File: src/app/genes/genes-list/genes-list.component.ts [hunk 3]
```diff
@@ -110,54 +111,38 @@ export class GenesListComponent implements OnInit {
...
             this.totalRecords = (data.totalRecords) ? data.totalRecords : 0;
-
-            // Starts table with the nominations columns sorted in descending order
-            this.customSort({
-                data: this.dataSource,
-                field: 'nominations',
-                mode: 'single',
-                order: -1
-            } as SortEvent);
-
```
Comment (JessterB) [inline src/app/genes/genes-list/genes-list.component.ts:160]: I moved the default sort into the db query in api.ts

File: src/app/genes/gene-similar/gene-similar.component.ts [hunk 4]
```diff
@@ -201,17 +199,34 @@ export class GeneSimilarComponent implements OnInit {
...
+                        de.safety_rating_display_value = de.druggability[0].safety_bucket + ': ' +
+                            de.druggability[0].safety_bucket_definition;
+                        de.ab_modality_display_value = de.druggability[0].abability_bucket + ': ' +
+                            de.druggability[0].abability_bucket_definition;
+                    } else {
+                        de.pharos_class_display_value = this.noValue;
+                        de.sm_druggability_display_value = this.noValue;
+                        de.safety_rating_display_value = this.noValue;
+                        de.ab_modality_display_value = this.noValue;
+                    }
```
Comment (JessterB) [inline src/app/genes/gene-similar/gene-similar.component.ts:229]: Rather than trying to get sorts working with subfields, I used the displayValue approach that was used in genes-list

File: src/app/genes/gene-similar/gene-similar.component.ts [hunk 5]
```diff
@@ -407,20 +377,21 @@ export class GeneSimilarComponent implements OnInit {
...
+                }
             } else {
                 result = (value1 < value2) ? -1 : (value1 > value2) ? 1 : 0;
             }
 
             return (event.order * result);
         });
     }
-
-    getSortColumnField(index: number, col: any) {
```
Comment (JessterB) [inline src/app/genes/gene-similar/gene-similar.component.ts:419]: this method wasn't being used

File: src/app/genes/genes-list/genes-list.component.html [hunk 6]
```diff
@@ -116,7 +116,7 @@ <h2 class="vmi-wrapper full-width no-margin-top small-margin-left">
             </tr>
         </ng-template>
         <ng-template pTemplate="body" let-rowData let-columns="columns">
-            <tr [pSelectableRow]="rowData" tabindex="0">
+            <tr [pSelectableRow]="rowData">
```
Comment (JessterB) [inline src/app/genes/genes-list/genes-list.component.html:119]: I couldn't figure out what this tabindex was for, similar-genes didn't have it, and removing it didn't seem to change any behaviors...but I'm open to feedback here!

File: src/server/routes/api.ts [hunk 7]
```diff
@@ -185,7 +203,7 @@ connection.once('open', () => {
 
     GenesInfo.find({ nominations: { $gt: 0 } }).lean()
         .select(tableGenesByIdFields)
-        .sort({ hgnc_symbol: 1 })
+        .sort({ nominations: -1, hgnc_symbol: 1 })
```
Comment (JessterB) [inline src/server/routes/api.ts:206]: the default sort order for NT table

File: src/server/routes/api.ts [hunk 8]
```diff
@@ -629,15 +647,16 @@ connection.once('open', () => {
...
     router.get('/mgenes/infos', (req, res, next) => {
         // Return an empty array in case no id was passed or no params
         if (!req.query || !req.query.ids) {
             res.json({ items: [] });
         } else {
             const ids = req.query.ids.split(',');
 
-            GenesInfo.find({ ensembl_gene_id: { $in: ids } }).lean().exec(
+            GenesInfo.find({ ensembl_gene_id: { $in: ids } }).lean()
+                .sort({  hgnc_symbol: 1 }).select(tableSimilarGenesFields).exec(
```
Comment (JessterB) [inline src/server/routes/api.ts:659]: Added a default sort here for the SG table

File: src/app/genes/gene-similar/gene-similar.component.ts [hunk 9]
```diff
@@ -67,19 +68,16 @@ export class GeneSimilarComponent implements OnInit {
...
-            { field: 'druggability', subfield: 'sm_druggability_bucket',
-                header: 'Small Molecule Druggability'},
-            { field: 'druggability', subfield: 'safety_bucket',
-                header: 'Safety Rating'},
-            { field: 'druggability', subfield: 'abability_bucket',
-                header: 'Antibody Modality'}
+            { field: 'pharos_class_display_value', header: 'Pharos Class' },
+            { field: 'sm_druggability_display_value', header: 'Small Molecule Druggability' },
+            { field: 'safety_rating_display_value', header: 'Safety Rating' },
+            { field: 'ab_modality_display_value', header: 'Antibody Modality' },
```
Comment (JessterB) [inline src/app/genes/gene-similar/gene-similar.component.ts:80]: Instead of using subfields here, I'm using the display field approach from genes-list.

File: src/app/genes/genes-list/genes-list.component.html (hunk 6 above)
Comment (nickgros) [reply src/app/genes/genes-list/genes-list.component.html:119]: If it doesn't impact usability for non-mouse users then I don't have a problem with removing it.

File: src/app/genes/gene-similar/gene-similar.component.ts [hunk 10]
```diff
@@ -231,33 +246,24 @@ export class GeneSimilarComponent implements OnInit {
...
                 downloadArray[downloadArray.length - 1] += ((index) < (this.cols.length - 1)) ?
                     ',' : '';
             });
         });
 
         // Finally export to csv
-        this.dataService.exportToCsv(
-            'genes-similar-' + this.gene.ensembl_gene_id + '.csv', downloadArray
-        );
+        this.dataService.exportToCsv( 'genes-similar-' + this.gene.ensembl_gene_id + '.csv', downloadArray);
```
Comment (nickgros) [inline src/app/genes/gene-similar/gene-similar.component.ts:266]: Just wanted to note that it's a little strange to me that we handle building and escaping this CSV client-side, but that's outside of the scope of this PR

**Review by nickgros — APPROVED** [review]
LGTM
